Diagnose Data for Cleaning

Veride bazı temiz olmayan şeyler olabilir ve bunları teşhis edip temizlememiz gerekir. Temiz olmayan şeyler şunlardır:

-> Sütun isimlerinde büyük-küçük harf veya kelimeler arasında boşluk gibi tutarsızlıkların olması
        Mesela sütun ismi 'Type 1' ve biz bu sütunu data.Type1 şeklinde çağırmak isterdik fakat arada boşluk olduğu için ve 
        data.Type 1 şeklinde hata alacağımızdan bu durum kod yazarken kısıtlanmamıza sebep olur. Bu şekilde çağıramadığımız 
        için data["Type 1"] şeklinde çağırmak zorundayız. "Type 1" sütununun "type1" şeklinde küçük harfle ve boşluk olmadan 
        isimlendirilmesi tercihimiz.

-> Çoğu veri setinde missing data bulunur. Missing data şudur: Bir sütuna ait verinin '-' veya 'nan' olmasıdır. Geçerli bir veri değildir.

-> Data'nın anlaşılabilirliği açısından veri setinde kullanılan dilin anlaşılabilir olması lazım. Mesela veri setinde farklı dillerden ifadeler bulunmaması gerekir.


Veriyi incelerken ilk olarak head, tail, columns, shape, info metodlarından yardım alıyoruz. Bu metodları uygulayalım. Öncelikle tabiki pandas kütüphanesini import edelim ve veri setimizi çağıralım.

In [1]:
import pandas as pd

data = pd.read_csv("pokemon.csv")

In [2]:
data.head()

#           Name Type 1  Type 2  HP  Attack  Defense  Sp. Atk  Sp. Def  \
0  1      Bulbasaur  Grass  Poison  45      49       49       65       65   
1  2        Ivysaur  Grass  Poison  60      62       63       80       80   
2  3       Venusaur  Grass  Poison  80      82       83      100      100   
3  4  Mega Venusaur  Grass  Poison  80     100      123      122      120   
4  5     Charmander   Fire     NaN  39      52       43       60       50   

   Speed  Generation  Legendary  
0     45           1      False  
1     60           1      False  
2     80           1      False  
3     80           1      False  
4     65           1      False

head() metodu default olarak ilk 5 sample'ı (satırı) gösterir. Bu metod ile veri setinin içeriği hakkında fikir sahibi olabiliriz.

In [3]:
data.tail()

#            Name   Type 1 Type 2  HP  Attack  Defense  Sp. Atk  \
795  796         Diancie     Rock  Fairy  50     100      150      100   
796  797    Mega Diancie     Rock  Fairy  50     160      110      160   
797  798  Hoopa Confined  Psychic  Ghost  80     110       60      150   
798  799   Hoopa Unbound  Psychic   Dark  80     160       60      170   
799  800       Volcanion     Fire  Water  80     110      120      130   

     Sp. Def  Speed  Generation  Legendary  
795      150     50           6       True  
796      110    110           6       True  
797      130     70           6       True  
798      130     80           6       True  
799       90     70           6       True

tail() metodu default olarak son 5 sample'ı (satırı) gösterir. Bu metod ile veri setinin içeriği hakkında fikir sahibi olabiliriz. Mesela 800 tane satır olduğunu tail() metodunun çıktısından anlayabiliyoruz.

In [4]:
data.columns

Index(['#', 'Name', 'Type 1', 'Type 2', 'HP', 'Attack', 'Defense', 'Sp. Atk',
       'Sp. Def', 'Speed', 'Generation', 'Legendary'],
      dtype='object')

Bir üst satır ile veri setindeki sütunları yani feature'leri elde ettik.

In [5]:
data.shape

(800, 12)

Bir üst satırın çıktısında, 800 tane satır yani sample ve 12 tane sütun yani feature olduğunu anladık.

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 800 entries, 0 to 799
Data columns (total 12 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   #           800 non-null    int64 
 1   Name        799 non-null    object
 2   Type 1      800 non-null    object
 3   Type 2      414 non-null    object
 4   HP          800 non-null    int64 
 5   Attack      800 non-null    int64 
 6   Defense     800 non-null    int64 
 7   Sp. Atk     800 non-null    int64 
 8   Sp. Def     800 non-null    int64 
 9   Speed       800 non-null    int64 
 10  Generation  800 non-null    int64 
 11  Legendary   800 non-null    bool  
dtypes: bool(1), int64(8), object(3)
memory usage: 69.7+ KB


info() metodu veri seti ile ilgili bilgiler verir. info() metodundan elde ettiğimiz çıktıyı okuyalım.

-> Bu bir DataFrame

-> 800 tane entries yani sample'ı var.

-> 12 tane sütun var.

-> Sütunlarda kaç adet null olmayan veri olduğu hakkında bilgi verir. Mesela Type 2 isimli sütunda 414 tane null olmayan yani object(string) veri var ve bu durumda 800-414=386 tane null veri var. Yani Type 2 sütun değeri olmayan 386 veri var.

-> Sütunlardan 1 tanesi bool tipinde, 8 tanesi int tipinde ve 3 tanesi object(string) tipinde.

Sonuç olarak, bu metodlar yardımıyla veri seti hakkında bu şekilde bilgiler elde edebiliriz.